README.md

```diff
@@ -148,506 +96,126 @@ export const Label: React.FC<PropsWithChildren<LabelProps>> = ({
-
-```ts
-root({
-  "size": {
-      "initial": "small", // this is mandatory to have an intiial option to respect mobile first approach
-      "md": "large"
+  defaultVariants: {
+    size: "small",
   },
-  "fontWeight": "xxl" // this would not accept a responsive value as you didnt' define it in the responsiveVariants array
-})
-```
+});
 
-With our latest definition
-
-```ts
-"slots": { 
-  "root": /** @tw */ "text-blue-500",
-  "otherSlot": /** @tw */ "text-blue-500" // here our new slot
-},
-"variants": {
-    "size": {
-        "small": /** @tw */ "text-xs",
-        "large": {
-          root: /** @tw */ "text-7xl",
-          otherSlot: /** @tw */ "text-12xl"
-        }
-    },
-    "fontWeight": {
-        xxl: {
-            "root": /** @tw */ "font-extrabold"
-        }
-    }
-},
-"defaultVariants": {
-  "size": "small",
-  "fontWeight": "xxl"
-},
-"responsiveVariants": ["size"]
+// Usage
+const { root, header, body } = cardVariants({ size: "large" });
 ```
 
-This would produce a class string like
-
-`text-blue-500 text-xs md:text-7xl font-extrabold`
-
-As you see we do regroup the responsive values right after their initial values.
+## Compound Variants
 
-4) Complex usage 1. How one responsive work with Compound
+Compound variants allow you to apply styles when multiple variant conditions are met:
 
-here our definition with compound
-
-```ts
-"slots": { 
-  "root": /** @tw */ "text-blue-500",
-  "otherSlot": /** @tw */ "text-blue-500" // here our new slot
-},
-"variants": {
-    "size": {
-        "small": /** @tw */ "text-xs",
-        "large": {
-          root: /** @tw */ "text-7xl",
-          otherSlot: /** @tw */ "text-12xl"
-        }
-    },
-    "fontWeight": {
-        xxl: {
-            "root": /** @tw */ "font-extrabold"
-        }
-    }
-},
-"defaultVariants": {
-  "size": "small",
-  "fontWeight": "xxl"
-},
-"responsiveVariants": ["size"],
-"compoundVariants": [
+```tsx
+export const buttonVariants = compose({
+  // ... previous definitions ...
+  compoundVariants: [
     {
-      "conditions": {
-        "size": "large",
+      conditions: {
+        intent: "primary",
+        size: "large",
       },
-      "class": /** @tw */ "bg-red-500 text-blue-500"
+      class: "font-bold",
     },
-    {
-      "conditions": {
-        fontWeight: "xxl",
-        size: "small"
-      },
-      class: /** @tw */ "bg-gray-500 border-red-500"
-    }
-]
-```
-
-When we do have responsive props what we would do generally is creating an object to all breakpoints you have defined.
-
-Fill the values with their initial values if there is no other responsive props.
-
-So in our use case
-
-```ts
-root({
-  "size": {
-      "initial": "small", // this is mandatory to have an intiial option to respect mobile first approach
-      "md": "large"
-  },
-  "fontWeight": "xxl", // this would not accept a responsive value as you didnt' define it in the responsiveVariants array,
-  className: "awesome-class"
-})
+  ],
+});
```

Comment:
Same here, I'd add some JSX to better visualize how it works


```tsx
// Usage in a React component
import { twMerge } from "tailwind-merge";

export const Button: React.FC<{}> = () => {
  const { root } = buttonVariants;

  return (
    <button
      // class .font-bold will be applied as the condition is fulfilled.
      className={twMerge(
        root({
          intent: "primary",
          size: large,
        })
      )}
    >
      Go
    <button>
};
```

Reply:
good point